spanner/client.go

```diff
@@ -265,7 +270,14 @@ func NewClientWithConfig(ctx context.Context, database string, config ClientConf
 	if config.BatchTimeout == 0 {
 		config.BatchTimeout = time.Minute
 	}
-
+	once.Do(func() {
+		// Initialize json provider.
+		jsonProvider = jsoniter.Config{
```

Comment:
See also my comment below: This will be confusing for anyone who calls `NewClientWithConfig` first with `UseNumber: false` and then later again with `UseNumber: true`. The configuration appears to be for a single client, but is in reality a global variable. We need to change that so it is either clearer, or it is actually a client config. Possible solutions for that could be:

1. Add a global method that sets the preferred decode method. Clearly document that the value that is set will be used by all clients, including both clients that have already been created and clients that will be created in the future. Based on where the configuration option is needed, this is probably the best/only possible solution.
2. OR: Add overloaded functions that accept an argument that determines which decoding method should be used wherever possible (this would not have my preference).
3. OR (I don't think it is possible, but for completeness): Pass a reference to the config on to all objects that need to know, and use the value there.

Reply:
I pushed changes for approach 1.